compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -290,11 +294,16 @@ public SemanticAnalyzer(CompilerContext context) {
     }
 
     public BLangPackage analyze(BLangPackage pkgNode) {
+        this.rootPackage = getRootPackage(pkgNode.moduleContextDataHolder.descriptor());
         this.dlog.setCurrentPackageId(pkgNode.packageID);
         pkgNode.accept(this);
         return pkgNode;
     }
 
+    private PackageID getRootPackage(ModuleDescriptor moduleDesc) {
+        return new PackageID(names.fromString(moduleDesc.org().value()),
+                names.fromString(moduleDesc.packageName().value()), names.fromString(moduleDesc.version().toString()));
```

Comment:
Instead of creating a new `PackageID`, we can just extract the information we need, right? Given that this analysis happens only via the package node, if we can extract the entire analysis out to a single method, I don't think we need a class field for this too.